handler/src/main/java/io/netty/handler/ssl/SslContextMapping.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import io.netty.util.AsyncMapping;
+import io.netty.util.Supplier;
+
+/**
+ * A map of hostnames to {@link Supplier}s of {@link SslContext}s.
+ *
+ * @see SniHandler
+ */
+public interface SslContextMapping extends AsyncMapping<String, Supplier<? extends SslContext>> {
```

Comment:
Do you think different supplier types would be co-mingled or even co-exist? I would assume most users use either JDK or OpenSSL, but not both. I'm wondering if it is practically useful from the user's perspective. If not then perhaps we can get rid of these generics. If there is not a concrete use case I would prefer to just use `Supplier<SslContext>` for now to keep things simple. We can mark the interfaces as unstable and add the wildcards later if there is no immediate use for them.